java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -379,6 +379,29 @@ public byte getByte(long index) {
     return offHeap.data.getByte(index * type.sizeInBytes);
   }
 
+  /**
+   * Get the byte list at index.
+   */
+  public byte[] getByteList(long index) {
+    assert type == DType.LIST;
+    assertsForGet(index);
+
+    int start = offHeap.offsets.getInt(index * DType.INT32.getSizeInBytes());
+    int end = offHeap.offsets.getInt((index + 1) * DType.INT32.getSizeInBytes());
+    int size = end-start;
+
+    // check if null or empty
+    if (start == end) {
+      if (isNull(index)) {
+        return null;
+      }
+    }
+
+    byte[] byteList = new byte[size];
+    offHeap.data.getBytes(byteList, 0, start, size);
```

Comment:
I thought that list types were going to have the data be in a child of the appropriate type so that it can be nullable.

@kuhushukla am I missing something here?

Reply:
No that is correct Bobby, @rwlee could u use the data from the child column? We had some bugs around this that were addressed but just like cudf cpp side, the data for a nested type sits in the child column. If u move this getByteList To the core class, u can recurse down the children.

Reply:
@rwlee ,  This is what `getList()` looks like that u can use to model `getByteList()`, and shows how data is picked from the children/child.
```
  /**
   * WARNING: Strictly for test only. This call is not efficient for production.
   */
  List getList(long rowIndex) {
    assert rowIndex < rows;
    assert type == DType.LIST;
    List retList = new ArrayList();
    int start = offHeap.offsets.getInt(rowIndex * DType.INT32.getSizeInBytes());
    int end = offHeap.offsets.getInt((rowIndex + 1) * DType.INT32.getSizeInBytes());
    // check if null or empty
    if (isNull(rowIndex)) {
      return null;
    }
    for(int j = start; j < end; j++) {
      for (HostColumnVectorCore childHcv : children) {
        // lists have only 1 child
        retList.add(childHcv.getElement(j));
      }
    }
    return retList;
  }
```

Reply:
```
assert type == DType.LIST;
assert children.get(0).type == DType.INT8;
children.get(0).getData().getBytes(byteList.....);
```

Reply:
Just some sample code, which can be tweaked of course to not get a particular index etc.